routers/api/v1/repo/pull.go

```diff
@@ -876,6 +878,27 @@ func MergePullRequest(ctx *context.APIContext) {
 	}
 
 	log.Trace("Pull request merged: %d", pr.ID)
+
+	if form.DeleteBranchAfterMerge {
+		if err := repo_service.DeleteBranch(ctx.User, ctx.Repo.Repository, ctx.Repo.GitRepo, pr.HeadBranch); err != nil {
```

Comment:
After reflecting on https://github.com/go-gitea/gitea/pull/16049#discussion_r660996140, I think this is wrong.
This only works if the pr head branch is in the same repo.
We need to delete the branch on pr.HeadRepo instead of ctx.Repo.Repository.